core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1251,7 +1251,22 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   @deprecated("use AccumulatorV2", "2.0.0")
   def accumulator[T](initialValue: T)(implicit param: AccumulatorParam[T]): Accumulator[T] = {
-    val acc = new Accumulator(initialValue, param)
+    accumulator(initialValue, dataProperty = false)
+  }
+
+  /**
+   * Create an [[org.apache.spark.Accumulator]] variable of a given type, which tasks can "add"
+   * values to using the `+=` method. Only the driver can access the accumulator's `value`.
+   *
+   * @param dataProperty If the accumulator should avoid re-counting multiple evaluations on the
+   *                     same RDD/partition. This adds some additional overhead for tracking and
+   *                     is an experimental feature.
+   */
+  @deprecated("use AccumulatorV2", "2.1.0")
```

Comment:
what's the argument for adding it to the old api?  Though I see the temptation, since its a nice new feature and we want to support users who haven't migrated yet -- I really don't think it makes sense to be adding stuff to a deprecated api.  It will make users think its not really deprecated.  Plus you provide a nice carrot for moving to the new api ;)